src/core_ocean/shared/mpas_ocn_init_routines.F

```diff
@@ -308,6 +308,54 @@ subroutine ocn_init_routines_setup_sign_and_index_fields(meshPool)!{{{
 
    end subroutine ocn_init_routines_setup_sign_and_index_fields!}}}
 
+!***********************************************************************
+!
+!  routine ocn_init_routines_area_weights
+!
+!> \brief  set up area weighting
+!> \author Mark Petersen
+!> \date   June 2020
+!> \details
+!>  This routine initializes edgeAreaFractionOfCell
+!
+!-----------------------------------------------------------------------
+   subroutine ocn_init_routines_area_weights(meshPool, edgeAreaFractionOfCell)!{{{
+
+      type (mpas_pool_type), intent(in) :: meshPool
+      real (kind=RKIND), dimension(:,:), intent(inout) :: edgeAreaFractionOfCell
+
+      integer, dimension(:), pointer :: nEdgesOnCell
+      integer, dimension(:,:), pointer :: edgesOnCell
+      real (kind=RKIND), dimension(:), pointer :: dcEdge, dvEdge, areaCell
+
+      integer, pointer :: nCells, nEdges
+      integer :: iCell, iEdge, i, j, k
+
+      call mpas_pool_get_dimension(meshPool, 'nCells', nCells)
+      call mpas_pool_get_dimension(meshPool, 'nEdges', nEdges)
+
+      call mpas_pool_get_array(meshPool, 'nEdgesOnCell', nEdgesOnCell)
+      call mpas_pool_get_array(meshPool, 'edgesOnCell', edgesOnCell)
+      call mpas_pool_get_array(meshPool, 'dcEdge', dcEdge)
+      call mpas_pool_get_array(meshPool, 'dvEdge', dvEdge)
+      call mpas_pool_get_array(meshPool, 'areaCell', areaCell)
+
+      do iCell = 1, nCells
+         do i = 1, nEdgesOnCell(iCell)
+            iEdge = edgesOnCell(i, iCell)
+            ! edgeAreaFractionOfCell is the fractional area of cell iCell
+            ! encompassed by the triangle with edge iEdge connected to the cell
+            ! center of cell iCell. On a perfect planar hex mesh this is always
+            ! 1/6. This weighting is used to interpolate scalars from edges to
+            ! cell centers. Use 2*edgeAreaFractionOfCell to interpolate normal
+            ! vectors at edges to vector norms at cell centers."
+            edgeAreaFractionOfCell(i, iCell) = &
+               0.25_RKIND * dcEdge(iEdge) * dvEdge(iEdge) / areaCell(iCell)
```

Comment:
I've introduced a new variable, `edgeAreaFractionOfCell`, which is the fractional area of the triangle on that edge. I hope this name is clear, and I'm open to suggestions. Triangle is already used for the dual cells. Hopefully this will clarify the confusion we had on this PR, and may same some compute time, as this exact sequence appears many times per timestep for every cell.

Reply:
Thanks @mark-petersen.  I think this is a good variable name and good documentation of how to use it.

Reply:
I like this as well.  Thanks for adding this!